homeassistant/components/screenlogic/number.py

```diff
@@ -0,0 +1,90 @@
+"""Support for a ScreenLogic number entity."""
+import logging
+
+from screenlogicpy.const import BODY_TYPE, DATA as SL_DATA, EQUIPMENT, SCG
+
+from homeassistant.components.number import NumberEntity
+
+from . import ScreenlogicEntity
+from .const import DOMAIN
+
+_LOGGER = logging.getLogger(__name__)
+
+SUPPORTED_SCG_NUMBERS = (
+    "scg_level1",
+    "scg_level2",
+)
+
+
+async def async_setup_entry(hass, config_entry, async_add_entities):
+    """Set up entry."""
+    coordinator = hass.data[DOMAIN][config_entry.entry_id]
+    equipment_flags = coordinator.data[SL_DATA.KEY_CONFIG]["equipment_flags"]
+    if equipment_flags & EQUIPMENT.FLAG_CHLORINATOR:
+        async_add_entities(
+            [
+                ScreenLogicNumber(coordinator, scg_level)
+                for scg_level in coordinator.data[SL_DATA.KEY_SCG]
+                if scg_level in SUPPORTED_SCG_NUMBERS
+            ]
+        )
+
+
+class ScreenLogicNumber(ScreenlogicEntity, NumberEntity):
+    """Class to represent a ScreenLogic Number."""
+
+    def __init__(self, coordinator, data_key, enabled=True):
+        """Initialize of the entity."""
+        super().__init__(coordinator, data_key, enabled)
+        self._body_type = SUPPORTED_SCG_NUMBERS.index(self._data_key)
+
+    @property
+    def name(self) -> str:
+        """Name of the number."""
+        return f"{self.gateway_name} {self.sensor['name']}"
+
+    @property
+    def value(self) -> float:
+        """Return the current value."""
+        return self.sensor["value"]
+
+    @property
+    def max_value(self) -> float:
```

Comment:
We can use `self._attr_max_value` here since this shouldn't change

https://github.com/home-assistant/core/blob/dev/homeassistant/components/demo/number.py#L91

Reply:
https://github.com/home-assistant/core/pull/60872/commits/ebfbe55c2bb6e61e3f33aeca281b6656af6ecaa7